File: coffee-system-api/src/main/java/com/jesperancinha/coffee/system/concurrency/QueueCallable.java
package com.jesperancinha.coffee.system.concurrency;

import java.util.List;
import java.util.concurrent.Callable;
import java.util.concurrent.Future;

/**
 * Created by joao on 8-5-16.
 */
public interface QueueCallable extends Callable<Boolean> {
    void addSubmitResult(Future<Boolean> submitResult);

    void waitForCalls();

    List<Callable<Boolean>> getAllCallables();
}

File: jeorg-kotlin-apps/coffee-paradigms/coffee-system/src/main/kotlin/org/jesperancinha/coffee/system/api/concurrency/QueueCallable.kt
package org.jesperancinha.coffee.system.api.concurrency

import java.util.concurrent.Callable
import java.util.concurrent.Future

/**
 * Created by joao on 8-5-16.
 */
interface QueueCallable : Callable<Boolean> {
    fun addSubmitResult(submitResult: Future<Boolean>)
    fun waitForCalls()
    val allCallables: MutableList<Callable<Boolean>>
}